text/3872-crates-io-security.md

```diff
@@ -0,0 +1,149 @@
+- Feature Name: crates-io-security
+- Start Date: 2025-10-27
+- RFC PR: [rust-lang/rfcs#3872](https://github.com/rust-lang/rfcs/pull/3872)
+- Rust Issue: [rust-lang/rust#3872](https://github.com/rust-lang/rust/issues/3872)
+
+# Summary
+
+[summary]: #summary
+
+This RFC proposes that crates.io should provide insight into vulnerabilities and unsound
+API surface based on the RustSec advisory database.
+
+# Motivation
+
+[motivation]: #motivation
+
+One of the roles that crates.io serves for Rust developers is as a discovery mechanism for library
+packages. As such, it is important that users can quickly assess the quality of a given crate,
+including security considerations such as unsound code/API or known vulnerabilities.
+The RustSec advisory database is a curated database of security advisories for Rust crates,
+which tracks known vulnerabilities, unsound code, and maintenance status of crates.
+
+The Rust ecosystem has a culture of having smaller, focused crates with a clear purpose.
+As a result, many Rust projects have a large number of dependencies, which increases the
+risk of introducing problems in the final artifact via the supply chain of dependencies.
+
+We've seen an increasing number of security issues via transitive dependencies:
+
+- In 2024, some releases of the popular xz-utils package (written in C) contained
+  a [malicious backdoor] affecting OpenSSH servers running on the local system.
+- In 2025, a phishing campaign [targeted crates.io users] using the `rustfoundation.dev` domain
+  name to impersonate the Rust Foundation and steal maintainer's credentials.
+
+While known actively malicious crates would be deleted from crates.io by the responsible team,
```

Comment:
While i'm in favor of adding a security tab to crates.io, this reasoning isn't the best-- it lists 2 examples of incidents and then immediately discounts them?

I feel like `log4j` is a better example, even though it's not Rust? In that case, the vulnerability was unintentional, severe, and it was important to know which versions were vulnerable and which versions were fixed.

Reply:
That's a good point. Added a commit to rewrite this section.